src/main/java/com/pyruby/stubserver/StubServer.java

```diff
@@ -213,8 +217,21 @@ public void handle(String target, HttpServletRequest httpServletRequest, HttpSer
                 proxy.matches(httpServletRequest);
                 proxy.respond(httpServletResponse);
             } else {
-                nullResponse.respond(httpServletResponse);
+                reportError(httpServletRequest, httpServletResponse);
+            }
+        }
+
+        private void reportError(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws IOException {
+            StringBuilder message = new StringBuilder("No expectation matched for:\n");
+            message.append(httpServletRequest.getMethod()).append(" ").append(httpServletRequest.getRequestURI()).append("\n");
+            Enumeration headerNames = httpServletRequest.getHeaderNames();
+            while (headerNames.hasMoreElements()) {
+                String headerName = (String) headerNames.nextElement();
+                Enumeration headerValues = httpServletRequest.getHeaders(headerName);
+                message.append(header(headerName, headerValues)).append("\n");
             }
+            Expectation.CannedResponse nullResponse = new Expectation.CannedResponse(404, "text/plain", message.toString(), null);
```

Comment:
This will break a number of existing users, as it changes the default from a 200 to a 404. The mockito model follows the 'Return null for unmatched' which is where this library drew it's inspiration from.

I would favour a switch or config that specified a 'Fail on unmatched' as an alternative.